meson.build

```diff
@@ -0,0 +1,151 @@
+project('pyfmmlib', 'c', 'fortran',
+  version : '2023.1',
+  license: 'MIT',
+  meson_version: '>=0.64.0',
+  default_options : ['warning_level=2', 'fortran_std=legacy'],
+)
+
+# {{{ gather includes
+
+py_mod = import('python')
+py = py_mod.find_installation(pure: false)
+py_dep = py.dependency()
+
+# NOTE: meson does not like absolute paths, so we dance around it
+incdir_numpy = run_command(py,
+  [
+    '-c',
+    '''import os
+import numpy as np
+try:
+  incdir = os.path.relpath(np.get_include())
+except Exception:
+  incdir = np.get_include()
+print(incdir)
+  '''], check: true).stdout().strip()
+inc_numpy = include_directories(incdir_numpy)
+
+incdir_f2py = incdir_numpy / '..' / '..' / 'f2py' / 'src'
+inc_f2py = include_directories(incdir_f2py)
+
+# }}}
+
+# {{{ generate vec wrappers
+
+generate_wrappers = files('vec_wrappers.py')
+wrapper_vec = custom_target('wrapper_vec',
+  output: 'vec_wrappers.f90',
+  command: [py, generate_wrappers, '@OUTPUT@'])
+
+# }}}
+
+# {{{ generate f2py wrappers
+
+f2py_name = '_internal'
+f2py_module_c = f'@f2py_name@module.c'
```

Comment:
Wow, meson sure seems to have found a way to make f-strings worse?

Reply:
Yeah, they use the `@VARIABLE@` thing like cmake for some reason, so I guess they just thought in for a penny.. :cry: